branchLayout/impl/src/test/java/com/virtuslab/branchlayout/unit/BranchLayoutTestSuite.java

```diff
@@ -190,4 +190,67 @@ public void withBranchSlideOut_givenNonExistingBranch_noExceptionThrown() {
     // then no exception thrown
     Assert.assertTrue(result.getRootEntries().isEmpty());
   }
+
+  @Test
+  public void equalsGivesCorrectResults() {
```

Comment:
Coz it's hard to deduce by a quick look... is there a case where two layouts differ just by **order of children** of certain entry (they should still be equal)?

Reply:
Should they be equal? Right now I've defined `IBranchLayoutEntry` equality as name, customAnnotation and children (recursively) are ALL equal. And so for `IBranchLayout` equality means that all rootEntries are equal (in a sense of `IBranchLayoutEntry` equality).

So to answer, if one of the entries of a branch layout will have an extra child branch(es) then it will not be equal to a branch layout that does not have this child branch.

This branch layout: 

```
A
  AA
B
  BA
  BB
```

is not equal to this one:

```
A
  AA
B
  BA
```

Reply:
Oh! I don't mean the difference in **structure**, just the difference in **child order**:
```
A
  B
  C
```
vs
```
A
   C
   B
```
AFAICS your implementation of `equals` considers the two layouts equal thanks to sorting the children before comparison right?

Reply:
Oh, sorry. Yeah it sorts them by name so order should not be a problem, but I will add a test case for that.